### ISSUE TITLE
[BUG] backend index error if no domain-record is set

### ISSUE THREAD

**Issue by gerchomat** [TYPO3-Solr/ext-solr#2259]
If there is no domain-record set in backend, the indexing will fail.

**To Reproduce**
1. Don't set any domain-record in the backend
2. Only set domain by $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['solr']['sites'][1]['domains']

![typo3_cms_9_5_4_](https://user-images.githubusercontent.com/16203876/52950909-cbd3da00-3380-11e9-86a9-afde93898434.png)
![typo3_cms_9_5_4_-2](https://user-images.githubusercontent.com/16203876/52950925-cf676100-3380-11e9-8a30-455a43919d6a.png)

The error goes back to `$this->redirect('index')` in /var/www/web/typo3conf/ext/solr/Classes/Controller/Backend/Search/IndexQueueModuleController.php line 274

 - TYPO3 Version: [9.5.4]
 - EXT:solr Version: [9.0.0]

So i think this is a structural bug?
Any opinions about that?

**Comment by timohund** [comment]
Hi @gerchomat i have the same setup here, without that issue.

I have a few questiuons:
* Did you Re-Initialize your solr connections after configuring:
```
$GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['solr']['sites'][1]['domains'][]  = 'www.foobar.test'
```
* Is 1 the uid of your sites root page?
* Can maybe try to analyze why this domain is used? Basically, a ViewHelper of the core is used and maybe there are some side effects with the indexing

**Comment by gerchomat** [comment]
@timohund yes we have reinitialized the connection

I figured out that at the moment the indexer is calling itself the `$_SERVER['HTTP_HOST']` is empty.
If i defined a domain-record it isn't

**Comment by timohund** [comment]
@gerchomat This should be fixed with 9.0.2. No feedback provided by slack. Please re-open or create a new issue if the problem still exists.
